Source/WebKit/Shared/Cocoa/CoreIPCDictionary.h

```diff
@@ -58,6 +63,9 @@ class CoreIPCDictionary {
     {
     }
 
+    void createNSDictionaryIfNeeded() const;
+
+    mutable RetainPtr<NSDictionary> m_nsDictionary;
```

Comment:
Instead of adding this, can we make ValueType a HashMap or just do linear searches on small finite sets of values?  Having an object that has a cached other form like this leads to error-prone code.

Reply:
The first code I wrote for this was making it a HashMap instead.
The problem that creates is having to write general hashing functions for all possible CoreIPCNSCFObject variants, which is actually why this has always been a Vector<KeyValuePairs>.

I agree on the point of the cached object leading to error prone code in general but the specifics here matter:
Since the validation step always happens on the decode side, and since we always need the NSDictionary on the decode side, leveraging the native Obj-C hashing seemed quite preferable.